src/prebuild.ts

```diff
@@ -0,0 +1,232 @@
+import * as path from "node:path";
+import * as fs from "node:fs";
+import { generateCssText } from "@webstudio-is/react-sdk";
+import { findTreeInstanceIds } from "@webstudio-is/project-build";
+import * as baseComponentMetas from "@webstudio-is/sdk-components-react/metas";
+import * as remixComponentMetas from "@webstudio-is/sdk-components-react-remix/metas";
+
+import type { Data } from "@webstudio-is/react-sdk";
+import type { Instance, Build, Page, Prop } from "@webstudio-is/project-build";
+import type { Params } from "@webstudio-is/react-sdk";
+import type { Asset } from "@webstudio-is/asset-uploader";
+import { BUILD_DIR } from "./lib";
+const projectId = process.argv.slice(2)[0] ?? 'sitedata';
+
+const baseDir = path.join(process.cwd(), `./${BUILD_DIR}/`);
+const jsonDir = path.join(process.cwd(), `./${BUILD_DIR}/app`);
+
+const siteDataString = fs.readFileSync(
+    path.join(baseDir, `${projectId}.json`),
+    "utf-8"
+);
+
+const siteData: Data & { user?: { email: string | null } } =
+    JSON.parse(siteDataString);
+
+const { assets } = siteData;
+const fontAssets: Data["assets"] = [];
+
+for (const asset of assets) {
+    if (asset.type === "font") {
+        fontAssets.push(asset);
+    }
+}
+
+type ComponentsByPage = {
+    [path: string]: Set<string>;
+};
+type SiteDataByPage = {
+    [path: string]: {
+        page: Page;
+        build: Pick<Build, "props" | "instances">;
+        assets: Array<Asset>;
+        params?: Params;
+        pages: Array<Page>;
+    };
+};
+type RemixRoutes = {
+    routes: Array<{
+        path: string;
+        file: string;
+    }>;
+};
+const remixRoutes: RemixRoutes = {
+    routes: [],
+};
+
+const componentsByPage: ComponentsByPage = {};
+const siteDataByPage: SiteDataByPage = {};
+
+for (const page of Object.values(siteData.pages)) {
```

Comment:
this smells like a problem we will have to deal with every time we change something in the logic unless this is reused across all builds ... also for saas

Reply:
This entire file basically

Reply:
Yes, it is. If we change it in SaaS, we need to change it here. If we reuse from a shared thing (package), then it could fail every time we update the package.